### PR TITLE
User managing

### PR DESCRIPTION
**Pull request by maykremer**

### CODE REVIEW

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/common/InterviewerHeader/InterviewerHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
```
Comment (carmeltsadok) [inline src/common/InterviewerHeader/InterviewerHeader.jsx:1]: Remove file and folder. not in use

File: src/i18n.js [hunk 2]
```diff
@@ -20,13 +20,42 @@ const resources = {
         noInterviewersInNodeGroup: 'אין מראיינים במסלול',
         chooseNodeGroup: 'בחרו מסלול',
       },
+      headerTitles: {
+        interviewsSchedule: 'לו"ז ראיונות',
+        malshabData: 'נתוני מלש"ב',
+        appraiserPreperationKit: 'ערכת הכנה למעריך',
+        addUsers: 'הוספת משתמשים',
+        addUnit: 'הוספת יחידה',
+      },
```
Comment (carmeltsadok) [inline src/i18n.js:29]: Remove. not in use

File: src/i18n.js [hunk 3]
```diff
@@ -20,13 +20,42 @@ const resources = {
...
+        unitName: 'שם יחידה',
+        noSelectedUnit: 'אנא בחר יחידה',
+        noUnits: 'אין יחידות',
+        userName: 'שם משתמש',
+        password: 'סיסמה',
+        unit: 'יחידה',
+        noUsers: 'אין משתמשים',
+        unitNotAddWarning: 'יחידה לא נוספה',
+        userNotAddWarning: 'יוזר לא נוסף',
+      },
```
Comment (carmeltsadok) [inline src/i18n.js:48]: Separate to better claseses

File: src/stores/Units.store.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { makeAutoObservable } from 'mobx';
```
Comment (carmeltsadok) [inline src/stores/Units.store.js:1]: I think this file not in use

File: src/services/user.service.js [hunk 5]
```diff
@@ -11,6 +11,15 @@ class UserService {
     const { data } = await axios.get(`${config.uri.api}/api/user/${id}`, { headers });
     return data;
   }
+  static async getUsersByUnitId(unitId) {
+    const res = await axios.get(`${await config.uri.api}/api/user/${unitId}`).catch(() => {});
+    return res?.data;
+  }
```
Comment (carmeltsadok) [inline src/services/user.service.js:17]: This api route is incorrect. The `unitId` should be in the params, this is get user by id

File: src/services/unit.service.js [hunk 6]
```diff
@@ -11,6 +11,14 @@ class UnitService {
     const { data } = await axios.get(`${config.uri.api}/api/unit/${id}`, { headers });
     return data;
   }
+  static async getUnits() {
+    const res = await axios.get(`${await config.uri.api}/api/unit`).catch(() => {});
```
Comment (carmeltsadok) [inline src/services/unit.service.js:15]: remove catch. The catch will be when calling the function.

File: src/services/unit.service.js [hunk 7]
```diff
@@ -11,6 +11,14 @@ class UnitService {
...
+    return res?.data;
+  }
+  static async createUnit(unitName) {
+    const res = await axios.post(`${await config.uri.api}/api/unit`, { unitName }).catch(() => {});
```
Comment (carmeltsadok) [inline src/services/unit.service.js:19]: remove catch. The catch will be when calling the function.

File: src/services/user.service.js (hunk 5 above)
Comment (carmeltsadok) [reply src/services/user.service.js:17]: remove catch. The catch will be when calling the function.

File: src/services/user.service.js [hunk 8]
```diff
@@ -11,6 +11,15 @@ class UserService {
...
+
+  static async createUser(unitId, role, userName) {
+    const res = await axios.post(`${await config.uri.api}/api/user/`, { unitId, role, userName }).catch(() => {});
```
Comment (carmeltsadok) [inline src/services/user.service.js:20]: remove catch. The catch will be when calling the function.

File: src/pages/UserManaging/index.jsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  const [units, setUnits] = useState([]);
+  const [selectedUnit, setSelectedUnit] = useState('');
+  const [users, setUsers] = useState([]);
+
+  useEffect(async () => {
+    setUnits(await UnitService.getUnits());
+  }, []);
+
+  return (
+    <div className={classes.mainDiv}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/index.jsx:20]: This should be root

File: src/pages/UserManaging/index.jsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+          setUsers={setUsers}
+        />
+        <UnitDetails
+          unit={selectedUnit}
+          users={users}
+          setUsers={setUsers}
+        />
+      </div>
+      <AddUnit
+        className={classes.addUnitDialog}
```
Comment (carmeltsadok) [inline src/pages/UserManaging/index.jsx:37]: not in use

File: src/pages/UserManaging/index.styles.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+      width: '90vw',
+      alignItems: 'center',
+    },
+  },
+  mainDiv: {
+    display: 'flex',
+  },
+  paper: {
+    borderRadius: 15,
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManaging/index.styles.js:20]: not in use

File: src/pages/UserManaging/Units/Units.jsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const [displayedArray, setDisplayedArray] = useState(unitsArray);
+
+  useEffect(() => {
+    setDisplayedArray(unitsArray);
+  }, [unitsArray]);
+
+  return (
+    <DashboardCard className={classes.units}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/Units/Units.jsx:19]: Should be root

File: src/pages/UserManaging/UnitDetailsUsersTable/UnitDetailsUsersTable.styles.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    width: '3rem',
+    marginLeft: '2rem',
+    border: '0.2px solid lightgrey',
+    borderRadius: '0.5rem',
+  },
+  addDiv: {
+    display: 'flex',
+    width: '70%',
+    justifyContent: 'space-between',
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsUsersTable/UnitDetailsUsersTable.styles.js:45]: not in use

File: src/pages/UserManaging/UnitDetailsUsersTable/UnitDetailsUsersTable.styles.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+      backgroundColor: 'rgb(244,245,247)',
+    },
+    width: '0.5rem',
+  },
+  noUsersDiv: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsUsersTable/UnitDetailsUsersTable.styles.js:66]: Not in use

File: src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.jsx [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+          setOpenNewUsersDialog(true);
+        }
+      }).catch(() => {
+        toast(t('text.userNotAddWarning'));
+      });
+    }
+  };
+
+  return (
+    <TableRow key={roleToDisplay} className={classes.tableRow}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.jsx:50]: Should be root

File: src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js:7]: not in use

File: src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+    margin: '0px -2rem 0px 3rem',
+    fontWeight: 'bold',
+    paddingRight: '2rem',
+    paddingLeft: '2rem',
+  },
+  addButton: {
+    backgroundColor: theme.palette.primary.main,
+    color: 'white',
+    borderRadius: '20rem',
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js:28]: not in use

File: src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+    justifyContent: 'space-between',
+    [theme.breakpoints.down('md')]: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      width: '90%',
+    },
+  },
+  tableHeadLine: {
+    fontWeight: 'bold',
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsTableRow/UnitDetailsTableRow.styles.js:66]: not in use

File: src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+            src='home.png'
+            alt=''
+            height='23rem'
+            width='23rem'
+            className={classes.homeLogo}
+          />
+          <p className={classes.mainDivText}>{`${unitName} : `}</p>
+          <img
+            src='users.png'
+            alt=''
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx:20]:
```suggestion
            alt='user logo'
```

File: src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+
+const UnitDetailsHeadLine = ({ unitName, numberOfUnitUsers }) => {
+  const classes = useStyles();
+  return (
+    <div className={classes.root}>
+      <DashboardCard className={classes.mainDiv}>
+        <div className={classes.main}>
+          <img
+            src='home.png'
+            alt=''
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx:12]:
```suggestion
            alt='home logo'
```

File: src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+            height='23rem'
+            width='23rem'
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx:14]: add to class

File: src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+            className={classes.homeLogo}
+          />
+          <p className={classes.mainDivText}>{`${unitName} : `}</p>
+          <img
+            src='users.png'
+            alt=''
+            height='36rem'
+            width='36rem'
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetailsHeadLine/UnitDetailsHeadLine.jsx:22]: add to class

File: src/pages/UserManaging/UnitDetails/UnitDetails.jsx [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+const UnitDetails = ({ unit, users, setUsers }) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  useEffect(async () => {
+    setUsers(await UserService.getUsersByUnitId(unit.id));
+  }, [unit]);
+
+  return (
+    <div>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetails/UnitDetails.jsx:18]: no need for this div

File: src/pages/UserManaging/UnitDetails/UnitDetails.jsx [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+      <DashboardCard className={classes.unitDetails}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetails/UnitDetails.jsx:19]: should be root

File: src/pages/UserManaging/UnitDetails/UnitDetails.jsx [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+        { unit
+          ? (
+            <>
+              <div className={classes.root}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/UnitDetails/UnitDetails.jsx:23]: not such class

File: src/pages/UserManaging/SearchBar/SearchBar.jsx [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <div className={classes.root}>
+      <div className={classes.searchDiv}>
+        <div className={classes.searchLogoDiv}>
+          <img
+            src='search.png'
+            alt=''
+            height='15rem'
+            width='15rem'
+            className={classes.searchLogo}
```
Comment (carmeltsadok) [inline src/pages/UserManaging/SearchBar/SearchBar.jsx:20]: not such class

File: src/pages/UserManaging/SearchBar/SearchBar.jsx [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+  };
+  return (
+    <div className={classes.root}>
+      <div className={classes.searchDiv}>
+        <div className={classes.searchLogoDiv}>
+          <img
+            src='search.png'
+            alt=''
+            height='15rem'
+            width='15rem'
```
Comment (carmeltsadok) [inline src/pages/UserManaging/SearchBar/SearchBar.jsx:19]: add to class

File: src/pages/UserManaging/SearchBar/SearchBar.jsx [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+  const handleChange = (value) => {
+    { value === '' ? setDisplayedArray(unitsArray) : setDisplayedArray(unitsArray.filter((unit) => unit.name.includes(value))); }
+  };
+  return (
+    <div className={classes.root}>
+      <div className={classes.searchDiv}>
+        <div className={classes.searchLogoDiv}>
+          <img
+            src='search.png'
+            alt=''
```
Comment (carmeltsadok) [inline src/pages/UserManaging/SearchBar/SearchBar.jsx:17]:
```suggestion
            alt='search logo'
```

File: src/pages/UserManaging/SearchBar/SearchBar.jsx [hunk 29]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-lone-blocks */
+/* eslint-disable no-unused-expressions */
```
Comment (carmeltsadok) [inline src/pages/UserManaging/SearchBar/SearchBar.jsx:2]: check if needed. If so, tell me and we will see if needed to add to eslint file

File: src/pages/UserManaging/RoleUsersDialog/RoleUsersDialog.jsx [hunk 30]
```diff
@@ -0,0 +1,74 @@
...
+              {unit.name}
+            </Typography>
+            <Typography>|</Typography>
+            <Typography>
+              <strong className={classes.titles}>{t('text.role')}</strong>
+              {role}
+            </Typography>
+            <Typography>|</Typography>
+            <Typography>
+              <strong style={{ marginRight: '-1rem' }} className={classes.titles}>{t('text.amount')}</strong>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/RoleUsersDialog/RoleUsersDialog.jsx:46]: add `style={{ marginRight: '-1rem' }}` to class

File: src/pages/UserManaging/RoleUsersDialog/RoleUsersDialog.jsx [hunk 31]
```diff
@@ -0,0 +1,74 @@
...
+      <div className={classes.root}>
+        <div className={classes.titleDiv}>
+          <Typography className={classes.usersTitle}>{t('text.users')}</Typography>
+          <Button className={classes.closeButton} onClick={() => { setOpenUsersDialog(false); }}>
+            <img src='add-icon.svg' alt='close' className={classes.closeIcon} />
+          </Button>
+        </div>
+        <div className={classes.headLineMainDiv}>
+          <div className={classes.headLine}>
+            <Typography style={{ wordSpacing: '1rem' }}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/RoleUsersDialog/RoleUsersDialog.jsx:35]: add `style={{ wordSpacing: '1rem' }}` to class

File: src/pages/UserManaging/NewUsersTable/NewUsersTable.jsx [hunk 32]
```diff
@@ -0,0 +1,41 @@
...
+          </TableCell>
+          <TableCell align='center' className={classes.cell}>
+            {t('text.password')}
+          </TableCell>
+        </TableHead>
+        <TableBody>
+          {users.map((user) => (
+            <TableRow key={user.id}>
+              <TableCell align='center'>{user.name}</TableCell>
+              <TableCell align='center'>DFGJKL123456</TableCell>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/NewUsersTable/NewUsersTable.jsx:32]: what is it `DFGJKL123456` ?

File: src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx [hunk 33]
```diff
@@ -0,0 +1,90 @@
...
+          <Tooltip title={t('toolTip.copyUsers')}>
+            <IconButton onClick={() => { copyUsers(); }}>
+              <img src='copy.svg' alt='copy' />
+            </IconButton>
+          </Tooltip>
+          <Snackbar
+            open
+          >
+            <Alert severity='error' variant='filled' style={{ marginBottom: '15%' }}>
+              <Typography style={{ fontSize: '1.1rem' }}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx:78]: add styles to classes

File: src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx [hunk 34]
```diff
@@ -0,0 +1,90 @@
...
+              {unit.name}
+            </Typography>
+            <Typography>|</Typography>
+            <Typography>
+              <strong className={classes.titles}>{t('text.role')}</strong>
+              {role}
+            </Typography>
+            <Typography>|</Typography>
+            <Typography>
+              <strong style={{ marginRight: '-1rem' }} className={classes.titles}>{t('text.amount')}</strong>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx:61]: add style to class

File: src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx [hunk 35]
```diff
@@ -0,0 +1,90 @@
...
+      <div className={classes.root}>
+        <div className={classes.titleDiv}>
+          <Typography className={classes.usersTitle}>{t('text.newUsers')}</Typography>
+          <Button className={classes.closeButton} onClick={() => { closeDialog(); }}>
+            <img src='add-icon.svg' alt='close' className={classes.closeIcon} />
+          </Button>
+        </div>
+        <div className={classes.mainHeadLineDiv}>
+          <div className={classes.headLine}>
+            <Typography style={{ wordSpacing: '1rem' }}>
```
Comment (carmeltsadok) [inline src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx:50]: add style to class

File: src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx [hunk 36]
```diff
@@ -0,0 +1,90 @@
...
+  const closeDialog = () => {
+    setUsersToAdd([]);
+    setOpenNewUsersDialog(false);
+    setNumberOfUsersToAdd(0);
+  };
+  const copyUsers = () => {
+    const usernamesList = [];
+    users.forEach((user) => {
+      usernamesList.push(user.name);
+      usernamesList.push('DFGJKL123456');
```
Comment (carmeltsadok) [inline src/pages/UserManaging/NewUsersDialog/NewUsersDialog.jsx:27]: what is it `DFGJKL123456` ?

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/UserManagement/RoleUsersDialog/RoleUsersDialog.styles.js [hunk 37]
```diff
@@ -0,0 +1,83 @@
...
+    alignItems: 'center',
+    marginLeft: '1.5rem',
+    marginRight: '1.5rem',
+    borderRadius: 30,
+    width: '75%',
+  },
+  headLineMainDiv: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
```
Comment (carmeltsadok) [inline src/pages/UserManagement/RoleUsersDialog/RoleUsersDialog.styles.js:43]: not in use

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Comment by carmeltsadok** [comment]
Move all components to `components` folder @maykremer

File: src/i18n.js [hunk 38]
```diff
@@ -20,13 +20,40 @@ const resources = {
         noInterviewersInNodeGroup: 'אין מראיינים במסלול',
         chooseNodeGroup: 'בחרו מסלול',
       },
+      headerTitles: {
+        addUsers: 'הוספת משתמשים',
+        addUnit: 'הוספת יחידה',
+      },
+      userManagement: {
```
[the remark this answers is not in the document]
Comment (carmeltsadok) [reply src/i18n.js:27]: Remove `userManagement` field. Add the other fields into the other existing fields

File: src/pages/UserManagement/components/NewUsersTable/NewUsersTable.jsx [hunk 39]
```diff
@@ -0,0 +1,41 @@
...
+import { useTranslation } from 'react-i18next';
+import useStyles from './NewUsersTable.styles.js';
+
+const NewUsersTable = ({ users }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  return (
+    <TableContainer component={Paper} className={classes.root}>
+      <Table aria-label='simple table' dir='rtl'>
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/NewUsersTable/NewUsersTable.jsx:19]: Is `dir='rtl'` required?

File: src/pages/UserManagement/components/NewUsersTable/NewUsersTable.styles.js [hunk 40]
```diff
@@ -0,0 +1,16 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(() => ({
+  root: {
+    boxShadow: 'none',
+    paddingRight: '2rem',
+    width: '60%',
+    direction: 'rtl',
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/NewUsersTable/NewUsersTable.styles.js:8]: is required?

File: src/pages/UserManagement/components/RolesDialogsHeadLine/RolesDialogsHeadLine.jsx [hunk 41]
```diff
@@ -0,0 +1,32 @@
...
+        <Typography>|</Typography>
+        <Typography>
+          <strong className={classes.titles}>{t('userManagement.titles.role')}</strong>
+          {role}
+        </Typography>
+        <Typography>|</Typography>
+        <Typography>
+          <strong className={classes.titles}>{t('userManagement.titles.amount')}</strong>
+          {users.length}
+        </Typography>
```
Comment (carmeltsadok) [inline src/pages/UserManagement/components/RolesDialogsHeadLine/RolesDialogsHeadLine.jsx:26]: You can do it in one `Typography`

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/i18n.js [hunk 42]
```diff
@@ -44,13 +79,24 @@ const resources = {
...
-        sizeLimitation: '( MB {{sizeLimit}} עד)',
+        sizeLimitation: '( עד {{sizeLimit}} MB )',
         hideFromView: 'הסתר מהתצוגה',
         fileInvalidType: 'סוג הקובץ לא תקין',
         fileTooLarge: 'הקובץ גדול מדי',
       },
+      roles: {
+        interviewer: 'מראיינ.ת',
+        ramadIturOfUnit: 'רמ"ד איתור',
+        ramadIturAssistant: 'עוזר רמ"ד איתור',
```
Comment (carmeltsadok) [inline src/i18n.js:92]:
```suggestion
        ramadIturAssistant: 'עוזר.ת רמ"ד איתור',
```

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Review by AdarPascal — APPROVED** [review]

**Pull request merged**